src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,518 @@
+     */
+    public static void addTaskFromFile(String taskDataString) {
+        Task task;
+        boolean isDone;
+        String[] words = taskDataString.split(" \\| "); //Length = 3 for Todo; 4 for Deadline, Event
+        try {
+            switch (words[1]) {
+            case "0":
+                isDone = false;
+                break;
+            case "1":
+                isDone = true;
+                break;
+            default: //if second letter (done status) is not 0 or 1
+                throw new InvalidCommandFormatException(MESSAGE_ERROR_INVALID_FILE_SYNTAX + LS + taskDataString);
+            }
+
+            switch (words[0]) {
+            case Task.TODO_ACRONYM:
+                task = new Todo(words[2]);
+                break;
+            case Task.DEADLINE_ACRONYM:
+                task = new Deadline(words[2], words[3]);
+                break;
+            case Task.EVENT_ACRONYM:
+                task = new Event(words[2], words[3]);
+                break;
+            default: //if first letter is not any of the valid task acronyms
+                throw new InvalidCommandFormatException(MESSAGE_ERROR_INVALID_FILE_SYNTAX + LS + taskDataString);
+            }
+            tasks.add(task);
+            if (isDone)  {
+                task.markAsDone();
+            }
+        } catch (IndexOutOfBoundsException e) {
+            showMessage(MESSAGE_ERROR_INVALID_FILE_SYNTAX, taskDataString, DIVIDER);
+        } catch (InvalidCommandFormatException e) {
+            showMessage(e.toString(), DIVIDER);
+        }
+    }
+
+    /**
+     * Append new data into task list file
+     *
+     * @param task Task to be appended to end of file
+     */
+    public static void appendNewTaskToFile(Task task) {
+        createFileIfDoesNotExist(true);
+        try {
+            FileWriter fw = new FileWriter(DATA_FILE_PATH, true);
+            fw.write(task.toTextFileString() + LS);
+            fw.close();
+        } catch (IOException e) {
+            showMessageFramedWithDivider(MESSAGE_ERROR_CANNOT_WRITE_TO_FILE);
+        }
+    }
+
+    /**
+     * Rewrite file with updated list of tasks
+     */
+    public static void rewriteTaskListToFile() {
+        createFileIfDoesNotExist(true);
+        try {
+            FileWriter fw = new FileWriter(DATA_FILE_PATH);
+            for (Task task : tasks) {
+               appendNewTaskToFile(task);
+            }
+            fw.close();
+        } catch (IOException e) {
+            showMessageFramedWithDivider(MESSAGE_ERROR_CANNOT_WRITE_TO_FILE);
+        }
+    }
+
+    /**
+     * Prints current tasks
+     */
+    public static void printTasks() {
+        if (tasks.isEmpty()) {
+            showMessageFramedWithDivider(MESSAGE_NO_TASKS_YET);
+        } else {
+            showMessage(DIVIDER, MESSAGE_INTRODUCE_TASKS);
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println(i+1 + ". " + tasks.get(i));
```

Comment:
Operators perhaps could be surrounded with whitespace: https://se-education.org/guides/conventions/java/index.html#layout
```suggestion
                System.out.println(i + 1 + ". " + tasks.get(i));
```